src/fields/FileField.cpp

```diff
@@ -0,0 +1,63 @@
+#include "FileField.h"
+
+using namespace std;
+
+FileField::FileField(string fieldName, string folderPath, string defaultValue, bool isPrimaryKey, bool isUnique, bool isNullable, bool isAutoFilled):
+    Field(fieldName, FILEFIELD, isPrimaryKey, isUnique, isNullable, isAutoFilled), _fieldValue(defaultValue), _defaultValue(defaultValue), _folderPath(folderPath) {}
+
+//Not sure if we should change this to setName or setFileName instead?
+void FileField::setValue(string value) {
+
+    //Note, in Djago they limit the file name length to 100 characters
+    setNull(false);
+
+    string pathToFile = _folderPath + "/" + _fieldValue;
+    string pathToNewFile = _folderPath + "/" + value;
+    rename(pathToFile.c_str(), pathToNewFile.c_str());
+
+    _fieldValue = value;
+}
+
+//Will save a copy of user-specified file to _folderPath directory
+//  File name will be stored in the database
+void FileField::saveNewFile(string pathToFile, string newFileName) {
+
+    ifstream inFile(pathToFile, ios::binary);
+
+    string newFilePath = _folderPath + "/" + newFileName;
+    ofstream newFile(newFilePath, ios::binary);
+    newFile << inFile.rdbuf();
+
+    _fieldValue = newFileName;
+}
+
+//Again, couldn't work out how to return a stream, so this just copies the
```

Comment:
Have you tried just making a void method like so:
```
void FileField::getFile(ifstream & fileStream) {
    string path= _folderPath + "/" + _fieldValue;
    fileStream(path, ios::binary);
}
```

Then in your main method, you would use it like:
```
ifstream fileStream;
m.shouldProbablyBeAPDBFile->getFile(fileStream);
\\ now do what you like with your file stream
```

In other words, you pass in a reference to the input stream, so you don't have to return it.